src/providers/twitch/IrcMessageHandler.cpp

```diff
@@ -707,12 +723,29 @@ void IrcMessageHandler::handleUserNoticeMessage(Communi::IrcMessage *message,
         else if (msgType == "announcement")
         {
             messageText = "Announcement";
+            QString announceCol = tags.value("msg-param-color").toString();
+            if (announceCol == "PRIMARY" || announceCol == "PURPLE" || announceCol == "")
+            {
+                announceFlag = MessageFlag::AnnouncePurple;
+            }
+            else if (announceCol == "BLUE")
+            {
+                announceFlag = MessageFlag::AnnounceBlue;
+            }
+            else if (announceCol == "GREEN")
+            {
+                announceFlag = MessageFlag::AnnounceGreen;
+            }
+            else if (announceCol == "ORANGE")
+            {
+                announceFlag = MessageFlag::AnnounceOrange;
+            }
         }
 
         auto b = MessageBuilder(systemMessage, parseTagString(messageText),
                                 calculateMessageTime(message).time());
 
-        b->flags.set(MessageFlag::Subscription);
+        b->flags.set(announceFlag);
```

Comment:
I believe that `MessageFlag::Subscription` is responsible for some other under-the-hood things like categorizing messages in search which perhaps shouldn't be touched.